File: pkg/models/ai_configuration.go
package models

// AIConfigurationUpdateRequest represents the owner-managed server AI settings.
// An empty API key keeps the currently persisted or environment-provided key.
type AIConfigurationUpdateRequest struct {
	Enabled        bool   `json:"enabled"`
	ImageEnabled   bool   `json:"imageEnabled"`
	BaseURL        string `json:"baseUrl" binding:"required,notBlank,max=500"`
	APIKey         string `json:"apiKey" binding:"omitempty,max=500"`
	ModelID        string `json:"modelId" binding:"required,notBlank,max=100"`
	EnableThinking string `json:"enableThinking" binding:"omitempty,max=10"`
	RequestTimeout uint32 `json:"requestTimeout" binding:"omitempty,min=1000,max=300000"`
}

// AIConfigurationResponse is safe to send to the browser and never includes a key.
type AIConfigurationResponse struct {
	Enabled          bool   `json:"enabled"`
	ImageEnabled     bool   `json:"imageEnabled"`
	Provider         string `json:"provider"`
	BaseURL          string `json:"baseUrl"`
	ModelID          string `json:"modelId"`
	EnableThinking   string `json:"enableThinking"`
	RequestTimeout   uint32 `json:"requestTimeout"`
	APIKeyConfigured bool   `json:"apiKeyConfigured"`
	Editable         bool   `json:"editable"`
}

File: src/models/ai_configuration.ts
export interface AIConfigurationResponse {
    readonly enabled: boolean;
    readonly imageEnabled: boolean;
    readonly provider: string;
    readonly baseUrl: string;
    readonly modelId: string;
    readonly enableThinking: string;
    readonly requestTimeout: number;
    readonly apiKeyConfigured: boolean;
    readonly editable: boolean;
}

export interface AIConfigurationUpdateRequest {
    readonly enabled: boolean;
    readonly imageEnabled: boolean;
    readonly baseUrl: string;
    readonly apiKey: string;
    readonly modelId: string;
    readonly enableThinking: string;
    readonly requestTimeout: number;
}
